File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/TimeSheet_ExportService/Models/time-sheet-export-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";

export class TimeSheet_ExportModel {
  year: number     = 0;
  month: number    = 0;
  selectedUserId: string[] = [];
  dal: string      = '';
  al: string       = '';
  idPar_ExportCau: number = 0;
}

export class TimeSheet_ExportInModel {
  timeSheet_Export: TimeSheet_ExportModel;
  constructor() {
    this.timeSheet_Export = new TimeSheet_ExportModel();
  }
}

export class TimeSheet_ExportOutModel extends ModelResult {
  timeSheet_Export: TimeSheet_ExportModel;
  constructor() {
    super();
    this.timeSheet_Export = new TimeSheet_ExportModel();
  }
}

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/TimeSheet_ExportService/Models/TimeSheet_ExportModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.TimeSheet_ExportService.Models
{
    
    public class TimeSheet_ExportModel
    {

        public int Year { get; set; }
        public int Month { get; set; } = 0;
        public List<string> SelectedUserId { get; set; } = new List<string>();
        public DateTime Dal { get; set; }
        public DateTime Al { get; set; }
        public int IdPar_ExportCau { get; set; }
        

    }

    public class TimeSheet_ExportInModel
    {
        public TimeSheet_ExportModel TimeSheet_Export { get; set; } = new TimeSheet_ExportModel();
    }
    public class TimeSheet_ExportOutModel : ModelResult
    {
        public TimeSheet_ExportModel TimeSheet_Export { get; set; } = new TimeSheet_ExportModel();

        public TimeSheet_ExportOutModel()
        {

        }
    }

    public class ExportRow
    {
        public string Cognome { get; set; } = string.Empty;
        public string Nome { get; set; } = string.Empty;
        public string Data { get; set; } = string.Empty;
        public string CodiceCausale { get; set; } = string.Empty;
        public decimal Valore { get; set; }
    }


}
